### PR TITLE
Preserve the color libraries are represented across benchmarks

### PR DESCRIPTION
**Pull request by brunocodutra**
This PR addresses #127.

The choice of colours may be seen in the charts below. 
Different datatypes of a given library were given the same hue but varying brightness.

hetero
type

### CODE REVIEW

File: benchmark/dependencies/hana.cmake [hunk 1]
```diff
@@ -29,7 +29,22 @@ if (METABENCH_HANA
     endif()
 
     function(Hana_add_dataset dataset)
-        metabench_add_dataset(${dataset} ${ARGN} MEDIAN_OF 3)
+        if(
```
Comment (ldionne) [inline benchmark/dependencies/hana.cmake]: Can't we use `${dataset} MATCHES "\\.(types|tuple|std_integer_sequence)"`.?

File: benchmark/dependencies/hana.cmake [hunk 2]
```diff
@@ -29,7 +29,22 @@ if (METABENCH_HANA
...
+        )
+            set(color "hsl(30, 100%, 50%)")
+        elseif(${dataset} MATCHES "\\.basic_tuple")
+            set(color "hsl(30, 100%, 62%)")
+        elseif(${dataset} MATCHES "\\.set")
+            set(color "hsl(30, 100%, 74%)")
+        elseif(${dataset} MATCHES "\\.map")
+            set(color "hsl(30, 100%, 86%)")
+        else()
+            message(FATAL_ERROR "Unknwon datatype ${dataset}")
```
Comment (ldionne) [inline benchmark/dependencies/hana.cmake]: Typo: __Unknown__

File: benchmark/dependencies/fusion.cmake [hunk 3]
```diff
@@ -9,7 +9,18 @@ if (METABENCH_FUSION AND NOT (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" AND
...
+            if(${dataset} MATCHES "\\.list")
+                set(color "hsl(292, 35%, 47%)")
+            elseif(${dataset} MATCHES "\\.vector")
+                set(color "hsl(292, 35%, 59%)")
+            elseif(${dataset} MATCHES "\\.set")
+                set(color "hsl(292, 35%, 71%)")
+            elseif(${dataset} MATCHES "\\.map")
+                set(color "hsl(292, 35%, 83%)")
+            else()
+                message(FATAL_ERROR "Unknwon datatype ${dataset}")
```
Comment (ldionne) [inline benchmark/dependencies/fusion.cmake]: Typo: __Unknown__

File: benchmark/dependencies/mpl.cmake [hunk 4]
```diff
@@ -6,14 +6,23 @@ if (METABENCH_MPL)
     find_package(Boost QUIET)
     if (Boost_FOUND)
         message(STATUS "Local MPL installation found - Boost ${Boost_VERSION}")
+        function(MPL_add_dataset dataset)
+            if(${dataset} MATCHES "\\.list")
+                set(color "hsl(292, 35%, 47%)")
+            elseif(${dataset} MATCHES "\\.vector")
+                set(color "hsl(292, 35%, 67%)")
+            else()
+                message(FATAL_ERROR "Unknwon datatype ${dataset}")
```
Comment (ldionne) [inline benchmark/dependencies/mpl.cmake]: Typo

File: benchmark/dependencies/std.cmake [hunk 5]
```diff
@@ -9,6 +9,13 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" AND CMAKE_CXX_COMPILER_VERSION VERSION
     endfunction()
 else()
     function(Std_add_dataset dataset)
-        metabench_add_dataset(${dataset} ${ARGN} MEDIAN_OF 3)
+        if(${dataset} MATCHES "\\.tuple")
+            set(color "hsl(118, 41%, 49%)")
+        elseif(${dataset} MATCHES "\\.array")
+            set(color "hsl(118, 41%, 69%)")
+        else()
+            message(FATAL_ERROR "Unknwon datatype ${dataset}")
```
Comment (ldionne) [inline benchmark/dependencies/std.cmake]: typo

File: benchmark/dependencies/hana.cmake (hunk 2 above)
Comment (brunocodutra) [inline benchmark/dependencies/hana.cmake]: good catch, thanks

File: benchmark/dependencies/hana.cmake (hunk 1 above)
Comment (brunocodutra) [inline benchmark/dependencies/hana.cmake]: will try

**Comment by ldionne** [comment]
This was merged in d5f64d9077aa1df619c1b44d33ecd009183fb2f8. I'm not sure why the PR did not autoclose. Thanks a lot!

**Pull request closed without merging**